Client/Assets/SessionRestore.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+ - License, v. 2.0. If a copy of the MPL was not distributed with this
+ - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+ <html>
+ <script>
+    (function () { 
```

Comment:
```
/**
 * This file is responsible for restoring session history.
 * It uses the DOM history API to push pages onto the back/forward stack. Since that API
 * is bound by same origin restrictions, we're only able to push pages with the current origin
 * (which is a page hosted on localhost). As a workaround, push all to-be-restored URLs as
 * error pages so that they will redirect to the correct URLs when loaded.
 */
```